.github/workflows/k8s-autodiscovery-test.yml

```diff
@@ -0,0 +1,42 @@
+name: k8s-autodiscovery-test
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 18 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  elastic-stack-snapshot-branches:
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    outputs:
+      matrix: ${{ steps.generator.outputs.matrix }}
+    steps:
+      - id: generator
+        uses: elastic/apm-pipeline-library/.github/actions/elastic-stack-snapshot-branches@current
+
+  k8s-autodiscovery-test:
+    needs: elastic-stack-snapshot-branches
+    strategy:
+      matrix: ${{ fromJson(needs.elastic-stack-snapshot-branches.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.branch }}
+      - id: get-go-version
+        name: Get Go version
+        run: echo "version=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ steps.get-go-version.outputs.version }}
```

Comment:
you can use go-version-file instead 

https://github.com/actions/setup-go/blob/main/action.yml#L7

Reply:
I was considering this.

But the version defined in https://github.com/elastic/e2e-testing/blob/6a560356cdeb0befde19147cbe1bcb75d88c407b/go.mod#L3

differs from the version defined in https://github.com/elastic/e2e-testing/blob/main/.go-version

and according to the docs it only accepts paths to `go.mod` or `go.work` files. Hence, I was not able to use `.go-version` file.